### PR TITLE
🎉 Source MSSQL: Added new comprehensive data type tests

### LINKED ISSUE #3794: Create comprehensive data type tests for MSSQL
## Tell us about the problem you're trying to solve
exact same as https://github.com/airbytehq/airbyte/issues/3562 but for MSSQL.

### PR DESCRIPTION
**Pull request by etsybaev**
## What
The current test coverage level is low and need to be increased as per https://github.com/airbytehq/airbyte/issues/3794

## How
Added comprehensive data type tests. A Follow up ticket for found bugs had been created https://github.com/airbytehq/airbyte/issues/4270

## Pre-merge Checklist
Expand the checklist which is relevant for this PR. 

<details><summary> <strong> Connector checklist </strong> </summary>
<p>

- [x] Issue acceptance criteria met
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [x] Secrets are annotated with `airbyte_secret` in output spec
- [x] Unit & integration tests added as appropriate (and are passing)
    * Community members: please provide proof of this succeeding locally e.g: screenshot or copy-paste acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] `/test connector=connectors/<name>` command as documented [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-an-existing-connector) is passing. 
    * Community members can skip this, Airbyters will run this for you. 
- [ ] Code reviews completed
- [ ] Credentials added to Github CI if needed and not already present. [instructions for injecting secrets into CI](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#using-credentials-in-ci). 
- [ ] Documentation updated 
    - [ ] README
    - [ ] CHANGELOG.md
    - [ ] Reference docs in the `docs/integrations/` directory.
    - [ ] Build status added to [build page](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/builds.md)
- [x] Build is successful
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [x] No major blockers
- [ ] PR merged into master branch
- [x] Follow up tickets have been created
- [ ] Associated tickets have been closed & stakeholders notified
</p>
</details>

<details><summary> <strong> Connector Generator checklist </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-mssql/src/test-integration/java/io/airbyte/integrations/source/mssql/MssqlSourceComprehensiveTest.java [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+            .sourceType("varbinary")
+            .fullSourceDataType("varbinary(30)")
+            .airbyteType(JsonSchemaPrimitive.STRING)
+            // .addInsertValues("CAST( 'ABC' AS VARBINARY)", "null")
+            // .addExpectedValues("A")
+            .addInsertValues("null")
+            .addNullExpectedValue()
+            .build());
+
+    // BUG: airbyte returns binary representation instead of readable one
```
Comment (yaroslav-hrytsaienko) [inline airbyte-integrations/connectors/source-mssql/src/test-integration/java/io/airbyte/integrations/source/mssql/MssqlSourceComprehensiveTest.java:332]: @sherifnada Should we create a follow up tickets to fix these kind of issues?

**Review by yaroslav-hrytsaienko — APPROVED** [review]

**Review by yaroslav-hrytsaienko — APPROVED** [review]

File: airbyte-integrations/connectors/source-mssql/src/test-integration/java/io/airbyte/integrations/source/mssql/MssqlSourceComprehensiveTest.java [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("tinyint")
+            .airbyteType(JsonSchemaPrimitive.NUMBER)
+            .addInsertValues("null", "0", "255")
+            .addExpectedValues(null, "0", "255")
+            .build());
+
+    // BUG Returns values true, false instead of 1 and 0.
```
[the remark this answers is not in the document]
Comment (etsybaev) [reply airbyte-integrations/connectors/source-mssql/src/test-integration/java/io/airbyte/integrations/source/mssql/MssqlSourceComprehensiveTest.java:138]: HI @yaroslav-hrytsaienko. I'd created a follow up ticket for all those fails https://github.com/airbytehq/airbyte/issues/4270

[the remark this answers is not in the document]
Comment (etsybaev) [reply airbyte-integrations/connectors/source-mssql/src/test-integration/java/io/airbyte/integrations/source/mssql/MssqlSourceComprehensiveTest.java:138]: @yaroslav-hrytsaienko added TODOss

**Review by yaroslav-hrytsaienko — APPROVED** [review]

File: airbyte-integrations/connectors/source-mssql/src/test-integration/java/io/airbyte/integrations/source/mssql/MssqlSourceComprehensiveTest.java [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("tinyint")
+            .airbyteType(JsonSchemaPrimitive.NUMBER)
+            .addInsertValues("null", "0", "255")
+            .addExpectedValues(null, "0", "255")
+            .build());
+
+    // TODO BUG Returns values true, false instead of 1 and 0.
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-mssql/src/test-integration/java/io/airbyte/integrations/source/mssql/MssqlSourceComprehensiveTest.java:138]: I think bit is used to express boolean in MSSQL so I think this is behaving as expected

https://stackoverflow.com/questions/1777257/how-do-you-create-a-yes-no-boolean-field-in-sql-server

**Review by sherifnada — APPROVED** [review]

File: airbyte-integrations/connectors/source-mssql/src/test-integration/java/io/airbyte/integrations/source/mssql/MssqlSourceComprehensiveTest.java (hunk 3 above)
Comment (etsybaev) [reply airbyte-integrations/connectors/source-mssql/src/test-integration/java/io/airbyte/integrations/source/mssql/MssqlSourceComprehensiveTest.java]: An official guide (https://docs.microsoft.com/en-us/sql/t-sql/data-types/bit-transact-sql?view=sql-server-ver15) says that the server converts true to 1 when you save it, but nothing about that stored 1 will be converted to true, that's why I considered it as a bug. Updated tests. made these 'true\false' as an expected behavior as you said. Thanks

Comment (tuliren) [reply airbyte-integrations/connectors/source-mssql/src/test-integration/java/io/airbyte/integrations/source/mssql/MssqlSourceComprehensiveTest.java]: It looks like `bit` columns are returned as `true` / `false` if there is no explicit conversion:

https://dbfiddle.uk/?rdbms=sqlserver_2019&fiddle=a2f618a30611ba2e995219b5fa24766d

Screen Shot 2021-06-25 at 07 47 10

**Pull request merged**